File: all-submissions/0252-meeting-rooms/solution.java
class Solution {
    public boolean canAttendMeetings(int[][] intervals) {
        if(intervals.length==0)
            return true;
//         int starts[] = new int[intervals.length];
//         int ends[] = new int[intervals.length];
//         for(int i = 0; i < intervals.length; i++){
//             starts[i] = intervals[i][0];
//             ends[i] = intervals[i][1];
//         }
        
//         Arrays.sort(starts);
//         Arrays.sort(ends);
        
        Arrays.sort(intervals, (a,b)->a[0]-b[0]);
        
        for(int i = 0; i<intervals.length-1;i++)
        {
          if(intervals[i][1]>intervals[i+1][0])
              return false;
            else
                continue;
        }
        return true;
    }
}

File: all-submissions/0252-meeting-rooms/solution.py
class Solution:
    def canAttendMeetings(self, intervals: List[List[int]]) -> bool:
        if not intervals:
            return True
        if len(intervals) == 0:
            return True
        intervals.sort(key = lambda x:x[0])
        end_time = intervals[0][1]
        for interval in intervals[1:]:
            if interval[0] < end_time:
                return False 
            end_time = interval[1]
        return True
